src/main/java/ui/TestController.java

```diff
@@ -157,4 +159,30 @@ public static RepoIO createTestingRepoIO(Optional<JSONStore> jsonStoreToBeUsed)
         return new RepoIO(Optional.of(new DummySource()), jsonStoreToBeUsed,
                           Optional.of(RepoStore.TEST_DIRECTORY));
     }
+
+    /**
+     * Create an update progress window if not in test mode.
+     *
+     * @return update progress window
+     */
+    public static UpdateProgressWindow createUpdateProgressWindow() {
+        if (isTestMode()) {
+            return null;
```

Comment:
Do we need to stub out this window in other places other than the method below? If not, a simple solution would be to just instantiate the window there.

Can those other tests be easily fixed, and when they are, does that solve your problem?